File: source/aoImportWizard/Models/ImportConfigModel.vb
﻿

Namespace Contensive.ImportWizard.Models
    Public Class ImportConfigModel
        Public Property importSource As ImportDataModel_ImportTypeEnum
        ''' <summary>
        ''' the uploaded data filename
        ''' </summary>
        ''' <returns></returns>
        Public Property privateUploadPathFilename As String
        '
        Public Property importMapPathFilename As String
        '
        Public Property notifyEmail As String
        ''' <summary>
        ''' contentid of the destination for the import
        ''' </summary>
        ''' <returns></returns>
        Public Property dstContentId As Integer
        '
        ''' <summary>
        ''' create new empty import map. ContentName requested because it must be valid before map is valid.
        ''' </summary>
        ''' <param name="app"></param>
        ''' <param name="contentName"></param>
        Public Sub newEmptyImportMap(app As ApplicationModel, contentName As String)
            If String.IsNullOrEmpty(contentName) Then
                '
                ' -- problem, cannot create map until content is established.
            End If
            'Dim mapFilenameData As New MapFilenameDataModel With {
            '    .mapName = "Import " & contentName,
            '    .dateCreated = Now
            '}
            importMapPathFilename = ImportMapModel.createMapPathFilename(app, contentName, "Import " & contentName & ".txt")
            save(app)
        End Sub
        '
        ''' <summary>
        ''' create the current instance of the visits importdataobject
        ''' </summary>
        ''' <param name="app"></param>
        ''' <returns></returns>
        Public Shared Function create(app As ApplicationModel) As ImportConfigModel
            Dim result As ImportConfigModel = app.cp.Visit.GetObject(Of ImportConfigModel)("ImportWizardData")
            '
            ' -- return data
            If result IsNot Nothing Then Return result
            '
            ' -- setup default values
            result = New ImportConfigModel With {
                .importSource = ImportDataModel_ImportTypeEnum.UploadFile,
                .notifyEmail = app.cp.User.Email,
                .dstContentId = 0
            }
            result.save(app)
            '
            ' -- setup a new import map
            'result.newEmptyImportMap(app)
            Return result
        End Function
        '
        ''' <summary>
        ''' save this instance of the data
        ''' </summary>
        ''' <param name="app"></param>
        Public Sub save(app As ApplicationModel)
            app.cp.Visit.SetProperty("ImportWizardData", app.cp.JSON.Serialize(Me))
        End Sub
        '
        ''' <summary>
        ''' clear the data
        ''' </summary>
        ''' <param name="app"></param>
        Public Shared Sub clear(app As ApplicationModel)
            app.cp.Visit.SetProperty("ImportWizardData", "")
        End Sub
        '
        '
        Public Shared Function convertImportTypeToInt(enumValue As ImportDataModel_ImportTypeEnum) As Integer
            Return Convert.ToInt32(enumValue)
        End Function
    End Class
    '
    Public Enum ImportDataModel_ImportTypeEnum
        UploadFile = 1
        SelectFile = 2

    End Enum
End Namespace

File: source/aoImportWizard/Models/ImportConfigModel.cs
using System;

namespace Contensive.ImportWizard.Models {
    public class ImportConfigModel {
        public ImportDataModel_ImportTypeEnum importSource { get; set; }
        /// <summary>
        /// the uploaded data filename
        /// </summary>
        /// <returns></returns>
        public string privateUploadPathFilename { get; set; }
        // 
        public string importMapPathFilename { get; set; }
        // 
        public string notifyEmail { get; set; }
        /// <summary>
        /// contentid of the destination for the import
        /// </summary>
        /// <returns></returns>
        public int dstContentId { get; set; }
        // 
        /// <summary>
        /// create new empty import map. ContentName requested because it must be valid before map is valid.
        /// </summary>
        /// <param name="app"></param>
        /// <param name="contentName"></param>
        public void newEmptyImportMap(ApplicationModel app, string contentName) {
            if (string.IsNullOrEmpty(contentName)) {
                // 
                // -- problem, cannot create map until content is established.
            }
            // Dim mapFilenameData As New MapFilenameDataModel With {
            // .mapName = "Import " & contentName,
            // .dateCreated = Now
            // }
            importMapPathFilename = ImportMapModel.createMapPathFilename(app, contentName, "Import " + contentName + ".txt");
            save(app);
        }
        // 
        /// <summary>
        /// create the current instance of the visits importdataobject
        /// </summary>
        /// <param name="app"></param>
        /// <returns></returns>
        public static ImportConfigModel create(ApplicationModel app) {
            var result = app.cp.Visit.GetObject<ImportConfigModel>("ImportWizardData");
            // 
            // -- return data
            if (result is not null)
                return result;
            // 
            // -- setup default values
            result = new ImportConfigModel() {
                importSource = ImportDataModel_ImportTypeEnum.UploadFile,
                notifyEmail = app.cp.User.Email,
                dstContentId = 0
            };
            result.save(app);
            // 
            // -- setup a new import map
            // result.newEmptyImportMap(app)
            return result;
        }
        // 
        /// <summary>
        /// save this instance of the data
        /// </summary>
        /// <param name="app"></param>
        public void save(ApplicationModel app) {
            app.cp.Visit.SetProperty("ImportWizardData", app.cp.JSON.Serialize(this));
        }
        // 
        /// <summary>
        /// clear the data
        /// </summary>
        /// <param name="app"></param>
        public static void clear(ApplicationModel app) {
            app.cp.Visit.SetProperty("ImportWizardData", "");
        }
        // 
        // 
        public static int convertImportTypeToInt(ImportDataModel_ImportTypeEnum enumValue) {
            return Convert.ToInt32((int)enumValue);
        }
    }
    // 
    public enum ImportDataModel_ImportTypeEnum {
        UploadFile = 1,
        SelectFile = 2

    }
}